openstack/networking/v2/extensions/vpnaas/services/requests.go

```diff
@@ -57,3 +57,39 @@ func Delete(c *gophercloud.ServiceClient, id string) (r DeleteResult) {
 	_, r.Err = c.Delete(resourceURL(c, id), nil)
 	return
 }
+
+// UpdateOptsBuilder allows extensions to add additional parameters to the
+// Update request.
+type UpdateOptsBuilder interface {
+	ToServiceUpdateMap() (map[string]interface{}, error)
+}
+
+// UpdateOpts contains the values used when updating a VPN service
+type UpdateOpts struct {
+	// Name is the human readable name of the service.
+	Name string `json:"name,omitempty"`
```

Comment:
Is it possible to _not_ have a Name? If so, let's say a user originally set a name but then wants to remove the name. By having the field configured like the above, that's not possible because an empty string is `string`'s zero-value and with `omitempty`, it won't be sent.

So we'll need to do:

```go
Name *string `json:"name,omitempty"`
```

That way a pointer to an empty string will be sent and successfully clear the name.